### ISSUE TITLE
Timezone would effect ZipConstants getFakeTime

### ISSUE THREAD

**Issue by weiyanlin117** [facebook/buck#2316]
When I am using buck http cache I am keep experiencing tons of warnings like:
```
error: XXXX debug map object file 'buck-out/gen/YYY.m.o' has changed (actual time is 1985-02-01 08:00:00.000000000, debug map time is 1985-02-01 00:00:00.000000000) since this executable was linked, file will be ignored
```
for every file I download from buck server.
Thus my build works fine but my breakpoint is keeping failure. 

After investigation, the root cause is here:

https://github.com/facebook/buck/blob/e1bfc4f4bf363d109c4654c784a32faa7ee2a192/src/com/facebook/buck/util/zip/ZipConstants.java#L42

`Calendar` create time would based on its environment local timezone,
for example 
```
import java.util.Calendar;
import java.util.TimeZone;

public class Main {

    public static void main(String[] args) {
        Calendar c = Calendar.getInstance();
        c.set(1985, Calendar.FEBRUARY, 1, 0, 0, 0);
        // my environment time is America/Los_Angeles
        System.out.println(c.getTimeInMillis());
        // 476092800837

        c.setTimeInMillis(476064000000L);
        System.out.println(c.getTimeInMillis());
        // 476064000000 
    }
}
```
 
https://github.com/facebook/buck/blob/e1bfc4f4bf363d109c4654c784a32faa7ee2a192/src/com/facebook/buck/util/zip/ZipConstants.java#L42
has no guarantee to be as same as 

https://github.com/facebook/buck/blob/a6c7d7391a7a8e575c6372cac6690a240b5907ea/src/com/facebook/buck/cxx/toolchain/objectfile/ObjectFileCommonModificationDate.java#L25

If cache timestamp is not consist, all cache consumer would experience the issue I mentioned at beginning.

This issue introduced from https://github.com/facebook/buck/commit/35c93b88c04ea88d1f65f149a73b7c549f38389a#diff-b12d580cc80ce3c8f0f5d2549fa61f67R4702

compare to `ZipEntry `, `TarArchiveEntry` would not include timezone convert.
https://docs.oracle.com/javase/8/docs/api/java/util/zip/ZipEntry.html#setTime-long-
https://commons.apache.org/proper/commons-compress/apidocs/org/apache/commons/compress/archivers/tar/TarArchiveEntry.html#setModTime-java.util.Date- 

Since `Calendar` class time would depended on timezone, we should not depend on Calendar generated unix timestamp, but use 
https://github.com/facebook/buck/blob/a6c7d7391a7a8e575c6372cac6690a240b5907ea/src/com/facebook/buck/cxx/toolchain/objectfile/ObjectFileCommonModificationDate.java#L25 directly as same as 
https://github.com/facebook/buck/blob/a6c7d7391a7a8e575c6372cac6690a240b5907ea/src/com/facebook/buck/cxx/FileLastModifiedDateContentsScrubber.java#L28

**Comment by weiyanlin117** [comment]
pr https://github.com/facebook/buck/pull/2317

**Comment by v-jizhang** [comment]
Fixed in #2317
